client/src/components/Upload/UploadBoxMixin.js

```diff
@@ -242,7 +242,8 @@ export default {
             } else {
                 this.counterAnnounce--;
             }
-            this.uploadbox.remove(model.id);
+            const fileSetKey = model.get("file_path") + model.get("file_size");
+            this.uploadbox.remove(model.id, fileSetKey);
```

Comment:
Do you we know why the `model.id` is not enough?

Reply:
[Here](https://github.com/galaxyproject/galaxy/pull/13652/files#diff-4943c3ffd7d0bc116b15b38b6e13ec2ebf9ec700284701e2a89733df17a3f04fR240) we check that selected file not existed in fileSet. Thus, if you select a file, then unselect it, it can't be selected anymore.

Reply:
see `client/src/utils/uploadbox.js:240`

Reply:
I see now, I think we need to make sure that `model.id` is returned properly when `uploadbox.add` is called and then pass the `id` back to the uploadbox.remove() as you do. Removing the fileSetKey should happen within the `uploadbox` module. Otherwise other location which also call `uplodbox.remove()` will also have to manually generate and parse a the fileSetKey.